src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1023,6 +1023,13 @@ private unsafe string ReplaceCore(string oldValue, string? newValue, CultureInfo
                 index = ci.IndexOf(this, oldValue, startIndex, this.Length - startIndex, options, &matchLength);
                 if (index >= 0)
                 {
+                    // There's the possibility that 'oldValue' has zero collation weight (empty string equivalent).
+                    // If this is the case, we need to fail immediately so that we don't enter an infinite loop.
+                    if (matchLength == 0)
+                    {
+                        throw new ArgumentException(SR.Argument_StringZeroLength, nameof(oldValue));
+                    }
```

Comment:
After seeing the customer scenario, it may be better to just return from the method without throwing. what you think?

Reply:
I thought about that, but then why not have string.Replace(string.Empty) also return the original string without throwing?

Reply:
> why not have string.Replace(string.Empty) also return the original string without throwing?

I am ok to make this not throw too. Throwing will require the callers to handle the exceptions and I am not expecting the callers will have the knowledge we can throw in this case.

Reply:
I'm also fine with making empty target strings not throw, as you had suggested. Would we be able to make this change or would it be considered too breaking?

Reply:
> Would we be able to make this change or would it be considered too breaking?

I think we can document the breaking changes for this case. The breaking scenario will be very limited as we are going not to throw in cases we used to throw before.